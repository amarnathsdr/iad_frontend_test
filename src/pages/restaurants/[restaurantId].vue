<script setup lang="ts">
import { useFetchRestaurant } from '~/composables/restaurants';
import { formattedPhoneNumber } from '~/helpers/formating';
import { averageRating } from '@/src/helpers/rating';

const { params } = useRoute();
const { data: restaurant, isError } = useFetchRestaurant({ restaurantId: params.restaurantId });
</script>

<template>
  <VAlert v-if="isError" type="warning" class="mt-4">
    Error occured while fetching the data
  </VAlert>
  <div v-else class="grid md:grid-cols-[minmax(0,_1fr)_16rem] gap-6">
    <VCard v-if="restaurant">
      <VImg
        v-for="photo in restaurant.photos"
        :key="photo"
        :src="restaurant.photos[0]"
        height="250"
        cover
        gradient="to top, rgba(0,0,0,.1), rgba(0,0,0,.5)"
      >
        <VCardTitle class="!text-4xl text-white">
          {{ restaurant.name }}
        </VCardTitle>
        <VRating
          class="mx-3"
          :model-value="averageRating(restaurant.reviews)"
          density="compact"
          readonly
        />
      </VImg>
      <VCardText>
        <div class="grid grid-cols-2 gap-4">
          <RestaurantLocation :location="restaurant.location" />
          <KeyValue icon="mdi-phone">
            <p class="text-body-1">
              {{ formattedPhoneNumber(restaurant.phone) }}
            </p>
          </KeyValue>
        </div>
      </VCardText>
    </VCard>
    <aside>
      <ul class="pa-0">
        <RestaurantReview :reviews="restaurant?.reviews" />
      </ul>
    </aside>
  </div>
</template>
