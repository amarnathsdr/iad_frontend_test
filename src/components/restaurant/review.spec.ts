import { mount } from '@vue/test-utils';

import RestaurantReview from './review.vue';
import { type Review } from '~/composables/restaurants';
import restaurantsResponse from '@/mock/restaurants.json';

const reviews = restaurantsResponse[0].reviews as Review[];

describe(`restaurant-review`, () => {
  test(`no props`, (context) => {
    const wrapper = mount(RestaurantReview, {
      global: { plugins: [context.router] },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it(`renders a review`, (context) => {
    const wrapper = mount(RestaurantReview, {
      global: { plugins: [context.router] },
      props: { reviews },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
