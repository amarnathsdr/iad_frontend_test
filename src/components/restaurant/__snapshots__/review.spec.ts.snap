// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`restaurant-review > no props 1`] = `
"<li class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated va-company-review\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <div class=\\"v-progress-linear__indeterminate\\">
        <div class=\\"v-progress-linear__indeterminate long\\"></div>
        <div class=\\"v-progress-linear__indeterminate short\\"></div>
      </div>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-text\\"></div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</li>"
`;

exports[`restaurant-review > renders a review 1`] = `
"<li class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated va-company-review\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <div class=\\"v-progress-linear__indeterminate\\">
        <div class=\\"v-progress-linear__indeterminate long\\"></div>
        <div class=\\"v-progress-linear__indeterminate short\\"></div>
      </div>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-text\\">
    <div class=\\"p-2\\">
      <div class=\\"v-rating v-rating--readonly v-theme--light\\"><label for=\\"v-rating-0-0\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 0 of 5</span>
          <!---->
        </label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-0\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"0\\">
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-1\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 1 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-1\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"1\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-2\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 2 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-2\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"2\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-3\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 3 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-3\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"3\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-4\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 4 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-4\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"4\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-5\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"5\\"></div>
          <!---->
        </div>
      </div>
      <div class=\\"flex-grow-1\\"></div><span>I've had decent lunches at Billy The Butcher a couple of times now. It's what I would describe as a trendy burger joint on the 4th floor of the Alsterhaus...</span>
    </div>
    <div class=\\"p-2\\">
      <div class=\\"v-rating v-rating--readonly v-theme--light\\"><label for=\\"v-rating-6-0\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 0 of 5</span>
          <!---->
        </label><input class=\\"v-rating__hidden\\" name=\\"v-rating-6\\" id=\\"v-rating-6-0\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"0\\">
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-6-1\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 1 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-6\\" id=\\"v-rating-6-1\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"1\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-6-2\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 2 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-6\\" id=\\"v-rating-6-2\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"2\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-6-3\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 3 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-6\\" id=\\"v-rating-6-3\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"3\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-6-4\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 4 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-6\\" id=\\"v-rating-6-4\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"4\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-6-5\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-6\\" id=\\"v-rating-6-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"5\\"></div>
          <!---->
        </div>
      </div>
      <div class=\\"flex-grow-1\\"></div><span>We arrived Hamburg at noon and decide to get some burgers before going to the office. This place is pretty close to the city hall and rated well.

I ordered...</span>
    </div>
    <div class=\\"p-2\\">
      <div class=\\"v-rating v-rating--readonly v-theme--light\\"><label for=\\"v-rating-12-0\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 0 of 5</span>
          <!---->
        </label><input class=\\"v-rating__hidden\\" name=\\"v-rating-12\\" id=\\"v-rating-12-0\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"0\\">
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-12-1\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 1 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-12\\" id=\\"v-rating-12-1\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"1\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-12-2\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 2 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-12\\" id=\\"v-rating-12-2\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"2\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-12-3\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 3 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-12\\" id=\\"v-rating-12-3\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"3\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-12-4\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 4 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-12\\" id=\\"v-rating-12-4\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"4\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-12-5\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-compact v-btn--size-default v-btn--variant-plain\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-12\\" id=\\"v-rating-12-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"5\\"></div>
          <!---->
        </div>
      </div>
      <div class=\\"flex-grow-1\\"></div><span>Nice place to have good burger for lunch. Most of personnel speak English which is good. Prices are decent, also I would recommend to share fries with a...</span>
    </div>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</li>"
`;
